File: algorithms/557. Reverse Words in a String III/solution.cc
class Solution {
public:
    string reverseWords(string s) {
        for ( int i = 0, pre = 0; i <= s.size(); ++i ) {
            if ( i == s.size() || s[i] == ' ' ) {
                reverse(s.begin() + pre, s.begin() + i);
                pre = i + 1;
            }
        }
        return s;
    }
};

// 30 / 30 test cases passed.
// Status: Accepted
// Runtime: 22 ms

File: algorithms/557. Reverse Words in a String III/solution.py
class Solution(object):
    def reverseWords(self, s):
        """
        :type s: str
        :rtype: str
        """
        return ' '.join(map(lambda x: x[::-1], s.split()))

# 30 / 30 test cases passed.
# Status: Accepted
# Runtime: 52 ms
